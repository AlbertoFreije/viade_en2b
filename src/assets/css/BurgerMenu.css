/* Position and sizing of burger button */
.bm-burger-button {
  margin-top: 5px;
  position: absolute;
  width: 46px;
  height: 30px;
  left: 28px;
  top: 36px;
  
  border-radius: 10px;
  background-color: var(--bg-color);
  box-shadow:  var(--box-shadow-up),
                var(--box-shadow-down);
}

/* Color/shape of burger icon bars */
.bm-burger-bars {
  background: var(--font-color);
  border-radius: 10px;
  margin: auto;
  width: 70%;
}

/* Color/shape of burger icon bars on hover*/
.bm-burger-bars-hover {
  background: var(--font-color-hover);
}

/* Position and sizing of clickable cross button */
.bm-cross-button {
  height: 24px;
  width: 24px;
}

/* Color/shape of close button cross */
.bm-cross {
  background: var(--font-color);
}

/*
Sidebar wrapper styles
Note: Beware of modifying this element as it can break the animations - you should not need to touch it in most cases
*/
.bm-menu-wrap {
  position: fixed;
  height: 100%;
}

/* General sidebar styles */
.bm-menu {
  background-color: var(--bg-color);
  padding: 2.5em 1.5em 0;
  font-size: 1.15em;
}

/* Morph shape necessary with bubble or elastic */
.bm-morph-shape {
  fill: var(--bg-color-pressed);
}

/* Individual item */
.bm-item {
  display: inline-block;
  color: var(--icon-color);
  font-size: 20px;
  padding: 10px;
  text-align: left;
  text-decoration:none;
}

.bm-item:hover, .link{
  color: var(--icon-color-hover);
  text-decoration: none;
}

.bm-item:focus, .link {
  color: var(--icon-color-hover);
  text-decoration: none;
}

/* Styling of overlay */
.bm-overlay {
  background: rgba(0, 0, 0, 0.3);
}

/* Text of the list items */
.bm-menu a {
    padding: 1em;
    text-transform: uppercase;
    display: block;
    color: var(--icon-color);
    transition: background 0.3s, box-shadow 0.3s;
    
    margin-top: 20px;
    margin-bottom: 20px;
    
    border-radius: 20px;
    
    box-shadow:  var(--box-shadow-up),
                  var(--box-shadow-down);
    
}
  
.bm-menu a:hover{
  color: var(--icon-color-hover);
  transition: 0.25s ease-out;
}

@keyframes colorTransition {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.bm-menu {
    height: 100%;
    padding: 0%;
    box-sizing: border-box;
    overflow: auto;
    background: var(--bg-color);
}

nav {
  display: block;
}

*, ::after, ::before {
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}

.menu-item:active {
  background-color: var(--bg-color-pressed);
}

.fa {
  width: 1.28571429em;
  display: inline-block;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.fa:focus {
  font-weight: 400;
}

.bm-item-list h2 span{
  color: var(--font-color);
}

.bm-item-list a span {
  margin-left: 10px;
  font-weight: 700;
  color: var(--font-color);
}

.bm-item-list a span:hover {
  color: var(--font-color-hover);
}


/* TOGGLE SWITCH - By W3Schools (lightly edited) - https://www.w3schools.com/howto/howto_css_switch.asp*/

/* The switch - the box around the slider */
.switch {
 position: relative;
 margin-left: 20px;
 display: inline-block;
 width: 50px;
 height: 1em;
}

/* Hide default HTML checkbox */
.switch input {
 opacity: 0;
 width: 0;
 height: 0;
}

/* The slider */
.slider {
 position: absolute;
 cursor: pointer;
 top: 0;
 left: 0;
 right: 0;
 bottom: 0;
 background-color: var(--primary-color);
 -webkit-transition: .4s;
 transition: .4s;
}

.slider:before {
 position: absolute;
 content: "";
 height: 13px;
 width: 13px;
 left: 4px;
 bottom: 4px;
 background-color: var(--bg-color);
 -webkit-transition: .4s;
 transition: .4s;
}

input:checked + .slider {
 background-color: var(--secondary-color);
}

input:checked + .slider:before {
 -webkit-transform: translateX(28px);
 -ms-transform: translateX(28px);
 transform: translateX(28px);
}

/* Rounded sliders */
.slider.round {
 border-radius: 34px;
}

.slider.round:before {
 border-radius: 50%;
}

/* Make the div look the same as the other menu items */

#lightmode {
  padding: 1em 1em 1em 2.30em;
  text-transform: uppercase;
  display: block;
  color: var(--icon-color);
  transition: background 0.3s, box-shadow 0.3s;
  margin-top: 20px;
  margin-bottom: 20px;
  border-radius: 20px;
  box-shadow: var(--box-shadow-up), var(--box-shadow-down);
}

div .menu-item :hover {
  color: var(--font-color-hover);
}

.toggle {
  margin-left: 10px;
  font-weight: 700;
  color: var(--font-color);
}
