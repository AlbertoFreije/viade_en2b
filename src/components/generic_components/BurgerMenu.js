import React, {Component} from 'react';
import { slide as Menu } from 'react-burger-menu';


import '../../assets/css/BurgerMenu.css'

class BurgerMenu extends Component {
  showSettings (event) {
    event.preventDefault();
  }

  render () {
    // NOTE: You also need to provide styles, see https://github.com/negomi/react-burger-menu#styling
    return (
    
      <Menu  disableAutoFocus className="bm-menu"  pageWrapId={ this.props.pageWrapId } outerContainerId={ this.props.container }>
        <nav className="bm-item-list">
        <h2 className="bm-item" tabIndex="0"><i className="fa fa-fw fa-inbox fa-2x"></i>
          <span>ViAde</span>
        </h2>
            <a id="home" className="menu-item" href="/">
              <i className="fa fa-fw fas fa-home"></i>
              <span>Home</span>
            </a>
            <a id="add-route" className="menu-item" href="/">
              <i className="fa fa-fw far fa-map"></i>
              <span>Add route</span>
            </a>
            <a id="list-routes" className="menu-item" href="/routes">
              <i className="fa fa-fw far fa-list-alt"></i>
              <span>List Routes</span>
            </a>
            <a id="list-friends" className="menu-item" href="/">
              <i className="fa fa-fw fas fa-users"></i>
              <span>List Friends</span>
            </a>
            <a id="about" className="menu-item" href="https://github.com/Arquisoft/viade_en2b">
              <i className="fa fa-fw fas fa-info-circle"></i>
              <span>About</span>
            </a>
            <a id="contact" className="menu-item" href="/">
              <i className="fa fa-fw fas far fa-address-card"></i>
              <span>Contact</span>
            </a>
            <a onClick={ this.showSettings } className="menu-item--small" href="/">
              <i className="fa fa-fw fas fa-cog"></i>
              <span>Settings</span>
            </a>
        </nav>
      </Menu>
    );
  }
}



export default BurgerMenu;
