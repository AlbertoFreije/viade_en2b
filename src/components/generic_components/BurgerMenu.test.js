import React, {Component} from 'react';
import BurgerMenu from './BurgerMenu';
import Enzyme, {shallow, mount} from "enzyme"
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});


test("DefineHamburger", ()=>{
    expect(BurgerMenu).toBeDefined();
});


test("RenderHamburger", ()=>{
    const temp = shallow(<BurgerMenu/>);
    expect(temp).toMatchSnapshot();
});


test("propsHamburgerEmptyPageWrap", ()=>{
    const temp = <BurgerMenu/>;
    expect(temp.props.pageWrapId).toBe(undefined);
});


test("propsHamburguerPageWrap", ()=>{
    const temp = <BurgerMenu 
        pageWrapId="page-wrap"
      />;
    expect(temp.props.pageWrapId).toBe("page-wrap");
});


test("propsHamburguerOuterContainer", ()=>{
    const temp = <BurgerMenu 
        container="outer-container"
      />;
    expect(temp.props.container).toBe("outer-container");
});


test("propsHamburguerAllProps", ()=>{
    const temp = <BurgerMenu
        pageWrapId="page-wrap"
        container="outer-container"
      />;
      expect(temp.props.pageWrapId).toBe("page-wrap");
      expect(temp.props.container).toBe("outer-container");
});


 test("HamburguerInexistentLink", () => {
  const temp = shallow(<BurgerMenu/>);
  expect(temp.find("#inexistentLink").exists()).toBe(false);
 });


test("HamburguerDefinedHome", () => {
  const temp = mount(<BurgerMenu/>);
  expect(temp.find("#home")).toBeDefined();
 });


test("HamburguerDefinedAddRoute", () => {
  const temp = mount(<BurgerMenu/>);
  expect(temp.find("#add-route")).toBeDefined();
 });


test("HamburguerDefinedListRoutes", () => {
  const temp = mount(<BurgerMenu/>);
  expect(temp.find("#list-routes")).toBeDefined();
 });


test("HamburguerDefinedListFriends", () => {
  const temp = mount(<BurgerMenu/>);
  expect(temp.find("#list-friends")).toBeDefined();
 });


test("HamburguerDefinedabout", () => {
  const temp = mount(<BurgerMenu/>);
  expect(temp.find("#about")).toBeDefined();
 });


test("HamburguerDefinedContact", () => {
  const temp = mount(<BurgerMenu/>);
  expect(temp.find("#contact")).toBeDefined();
 });


test("HamburguerDefinedSettings", () => {
  const temp = mount(<BurgerMenu/>);
  expect(temp.find("#settings")).toBeDefined();
 });


test("HamburguerClickHome", () => {
  const temp = mount(<BurgerMenu/>);
  expect(temp.find("#home").at(1).props().href).toBe("#/");
 });


test("HamburguerClickListRoutes", () => {
  const temp = mount(<BurgerMenu/>);
  expect(temp.find("#list-routes").at(1).props().href).toBe("#/routes");
 });


test("HamburguerClickAbout", () => {
  const temp = mount(<BurgerMenu/>);
  expect(temp.find("#about").at(1).props().href).toBe("#/");
 });
