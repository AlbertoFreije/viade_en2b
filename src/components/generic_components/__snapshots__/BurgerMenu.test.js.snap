// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderHamburger 1`] = `ShallowWrapper {}`;
